Makefile

```diff
@@ -646,6 +646,11 @@ test-geoserver: install-dev install		## run Geoserver requests tests against a c
 	@echo "Running local tests..."
 	@bash -c '$(CONDA_CMD) pytest tests -vv -m "geoserver" --junitxml "$(APP_ROOT)/tests/results.xml"'
 
+.PHONY: test-github
+test-github:  ## test target used by github's ci, runs all tests except Geoserver requests, without prior dependency check and installation
+	@echo "Running tests..."
+	@bash -c '$(CONDA_CMD) pytest tests -vv -m "not geoserver" --junitxml "$(APP_ROOT)/tests/results.xml"'
```

Comment:
I still don't understand the logic here. 
What is the point to skip geoserver tests? 
We want the test suite to validate that features still work each time we change something.

Reply:
`geoserver` might not be the best marker name for this, as it does not convey the online nature of this test suite. I had a chance to talk about this with @dbyrns and it was the solution we ended up with.

Those tests are only covering the specific requests to a Geoserver instance, hence they require a live Geoserver. I agree that this kind of tests  would be better suited for a notebook test at the birdhouse level like the other integration tests, but this was somewhat out of scope for this PR.

However, this series of tests were already written by that time and were quite useful while developing. Making Github run a Geoserver instance in it's workflow seemed a bit overkill (and would become redundant once the notebook tests are created) and transforming them to use a Mock reduces their effectiveness, as we wouldn't know if the requests are really valid from the tests. So I took the same approach as with Birdy that skips the `slow` and `online` tests in its Github workflow.

Reply:
Ok. I think a note about this specific reason (GeoServer instance online needed) would be important to add above the target, so we can remember about this after a few months. Maybe `online` marker would make this more explicit (the test could have both `geoserver` and `online`, and we skip `online` tests in CI which seems more natural).

I will leave it up to @dbyrns to take the final decision regarding if we should test something.
I still think that at least a minimal test should be added with a mock instance to ensure that the most basic operations work. We don't have to test all cases, just the most common workflow. I have found (with Weaver PR for example) that the full cycle of (1) develop component, (2) deploy in birdhouse, (3) test-notebooks, (4) oups! broken, (5) back to (1) is extremely tedious and time consuming. It's even worst when the broken thing is not even related to (1)'s feature but a side effect of it.

Reply:
I like the dual marker solution.

As for testing, right now the `test_request_task` class test our Geoserver service's user creation/removal, and file creation/removal implementations . I could use the same approach and patch the http requests so the CI tests reach all the way to the Geoserver requests. Less involved than mocking Geoserver, but combined with the `online` tests, it would give us a pretty good coverage at the component level.

It would only leave the shapefile publishing aspect untested, as it's squarely in the integration test category. Right now I test it manually (all the way to accessing it via WFS), but I think I could add it to the online tests with a bit of tinkering without it taking too long.

Reply:
I also like the dual marker solution. 
As for what should be tested, unlike Weaver, the Geoserver component is a small sub-system of Cowbird and should not evolve a lot. Mocking every services used by Cowbird would be very time consuming for the investment return. Given that we still have an online test that can validate the end-to-end functionnality, I think that's enough.

Reply:
Sounds fair. Make sure to apply `geoserver` to the relevant unit-tests that will run offline in the CI, this way we can run specifically those as needed.